Add bucket_script and bucket_selector pipeline aggregations. (#9738)
opensearch-project/documentation-website: _aggregations/pipeline/bucket-script.md

The `bucket_script` aggregation takes the following parameters:

| Parameter             | Required/Optional | Data type       | Description |
| :--                   | :--               |  :--            | :--         |
| `buckets_path`        | Required          | Object          | A map of variable names to bucketed metrics that identify the metrics to be used in the script. The metrics must be numeric. See [Script variables](#script-variables). |
| `script`              | Required          | String or Object | The script to execute. Can be an in-line script, stored script, or script file. The script has access to the variable names defined in the `buckets_path` parameter. Must return a numeric value. |
| `gap_policy`          | Optional          | String          | The policy to apply to missing data. Valid values are `skip`, `insert_zeros`, and `keep_values`. Default is `skip`. |
| `format`              | Optional          | String          | A [DecimalFormat](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/text/DecimalFormat.html) formatting string. Returns the formatted output in the aggregation's `value_as_string` property. |

## Script variables

> The `buckets_path` parameter maps script variable names to metrics from parent aggregations. These variables can then be used in the script. For example, the following `buckets_path` maps the `sales_sum` metric to the `total_sales` script variable and the `item_count` metric to the `item_count` script variable:

Author: ```suggestion
The `buckets_path` parameter maps script variable names to metrics from parent aggregations. These variables can then be used in the script. 

For the `bucket_script` and `bucket_selector` aggregations, the `buckets_path` parameter is an object rather than a string because it must refer to multiple bucket metrics. See the [Pipeline aggregations]({{site.url}}{{site.baseurl}}/aggregations/pipeline/index#buckets-path-property) page for a description of the string version of `buckets_path`.
{: .note}

The following `buckets_path` maps the `sales_sum` metric to the `total_sales` script variable and the `item_count` metric to the `item_count` script variable:
```

Reviewer: Do I still need to explicitly specify even when the mapping is identical, like for `item_count`?

Reviewer 2: My understanding is that you still need to explicitly specify the variable mapping in `buckets_path`, even if the variable name is identical to the aggregation name (e.g., `item_count`).

The `bucket_script` aggregation does not implicitly inherit variable names from existing aggregations or field mappings. It only knows what you explicitly pass in through `buckets_path`.

```json
"buckets_path": {
  "total_sales": "sales_sum",
  "item_count": "item_count"
}
```

The mapped variables can be accessed from the `params` context. For example:

- `params.total_sales`
- `params.item_count` 

## Enabling inline scripting

> Use the `script` parameter to add your script. The script can be inline, in a file, or in an index. To enable inline scripting, your `opensearch.yml` file in the `config` folder must contain the following:

Reviewer 3: ```suggestion
Use the `script` parameter to add your script. The script can be inline, in a file, or in an index. To enable inline scripting, the `opensearch.yml` file in the `config` folder must contain the following:
```

```yml
script.inline: on
```

## Example

> The following example creates a date histogram with a one-month interval from the OpenSearch Dashboards e-commerce sample data. The `total_sales` subaggregation sums the taxed price of all items sold for each month. The `vendor_count` aggregation counts the total number of unique vendors for the month. Finally, the `avg_vendor_spend` aggregation uses an inline script to calculate the average amount spent per vendor each month:

Reviewer 3: ```suggestion
The following example creates a date histogram with a one-month interval from the OpenSearch Dashboards e-commerce sample data. The `total_sales` subaggregation sums the taxed price of all items sold for each month. The `vendor_count` aggregation counts the total number of unique vendors for each month. Finally, the `avg_vendor_spend` aggregation uses an inline script to calculate the average amount spent per vendor each month:
```
